src/views/common/Animation/AnimationControls/useUrlPlaybackWindowSupport.ts

```diff
@@ -0,0 +1,69 @@
+import { useUrlState } from 'figurl/UrlStateContext'
+import React, { useCallback } from 'react'
+import { AnimationState, AnimationStateAction } from 'views/common/Animation/AnimationStateReducer'
+
+const useUrlPlaybackWindowSupport = (dispatch: React.Dispatch<AnimationStateAction<any>>) => {
```

Comment:
The goal here is to encapsulate interactions with the URL (an implementation detail that isn't the only way to handle the idea of bookmarking) and provide an abstract interface to the UI (which is why the bookmark button component only worries about 'bookmark this' or 'is this bookmarked' rather than referencing the URL directly).

We might need to flesh this interface out a little bit if we wanted to support removing the bookmarked state from the URL entirely (not something we generally do in the current model of URL query parameters) or if a different implementation needed to handle multiple possible bookmarks (e.g. local storage database).